app/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModel.scala

```diff
@@ -126,23 +106,55 @@ object IncomeSourceViewModel extends ViewModelHelper {
       } yield latestPayment.amountYearToDate
       ).getOrElse(0)
 
+    val amountString = if (amountNumeric != BigDecimal(0)) withPoundPrefixAndSign(MoneyPounds(amountNumeric, 0)) else ""
 
-    val amountString = if(amountNumeric != BigDecimal(0)) withPoundPrefixAndSign(MoneyPounds(amountNumeric, 0)) else ""
-
-    val endDate: Option[String] = employment.endDate.map( Dates.formatDate(_) )
+    val endDate: Option[String] = employment.endDate.map(Dates.formatDate(_))
 
     IncomeSourceViewModel(
       employment.name,
       amountString,
       "",
-      false,
+      displayTaxCode = false,
       employment.payrollNumber.getOrElse(""),
-      employment.payrollNumber.isDefined,
+      displayPayrollNumber = employment.payrollNumber.isDefined,
       endDate.getOrElse(""),
-      endDate.isDefined,
+      displayEndDate = endDate.isDefined,
       messages("tai.incomeTaxSummary.employment.link"),
-      controllers.routes.YourIncomeCalculationController.yourIncomeCalculationPage(employment.sequenceNumber).url,
-      true)
+      controllers.routes.YourIncomeCalculationController.yourIncomeCalculationPage(employment.sequenceNumber).url
+    )
+  }
+
+  def apply(incomeSource: IncomeSource)(implicit messages: Messages): IncomeSourceViewModel = {
```

Comment:
There is lots of logic here, shift it to a service and keep the ViewModel's simple

Reply:
in new service